src/App.tsx

```diff
@@ -17,18 +76,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={todos}
+                onStatusChanged={setFilterStatus}
+                onQueryChanged={handleInput}
+                status={filterStatus}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loaded ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodoId={selectedTodo?.id}
+                  onSelected={setSelectedTodo}
```

Comment:
The naming convention for event handler functions should be followed. Consider renaming `onSelected` to `handleSelect` or `onSelect` to align with the recommended naming conventions.